circuits/src/types/balance.rs

```diff
@@ -284,3 +299,125 @@ impl<N: MpcNetwork + Send, S: SharedValueSource<Scalar>> CommitVerifier
         })
     }
 }
+
+// ------------------------------
+// | Secret Shared Balance Type |
+// ------------------------------
+
+/// A balance that has been split into secret shares
+#[derive(Copy, Clone, Debug)]
+pub struct BalanceSecretShare {
+    /// The mint (ERC20 token addr) of the balance
+    pub mint: Scalar,
+    /// The amount of the balance held
+    pub amount: Scalar,
+}
+
+impl Add<BalanceSecretShare> for BalanceSecretShare {
+    type Output = Balance;
+
+    fn add(self, rhs: BalanceSecretShare) -> Self::Output {
+        let mint = scalar_to_biguint(&(self.mint + rhs.mint));
+        let amount = scalar_to_u64(&(self.amount + rhs.amount));
+
+        Balance { mint, amount }
+    }
+}
+
+impl BalanceSecretShare {
+    /// Apply a blinder to the secret shares
+    pub fn blind(&mut self, blinder: Scalar) {
+        self.mint += blinder;
+        self.amount += blinder;
+    }
+
+    /// Remove a blinder from the secret shares
+    pub fn unblind(&mut self, blinder: Scalar) {
+        self.mint -= blinder;
+        self.amount -= blinder;
+    }
+}
+
+/// A balance secret share that has been allocated in a constraint system
+#[derive(Copy, Clone, Debug)]
+pub struct BalanceSecretShareVar {
+    /// The mint (ERC20 token addr) of the balance
+    pub mint: Variable,
+    /// The amount of the balance held
+    pub amount: Variable,
+}
+
+impl Add<BalanceSecretShareVar> for BalanceSecretShareVar {
+    type Output = BalanceVar<LinearCombination>;
+
+    fn add(self, rhs: BalanceSecretShareVar) -> Self::Output {
+        BalanceVar {
+            mint: self.mint + rhs.mint,
+            amount: self.amount + rhs.amount,
+        }
+    }
+}
+
+impl BalanceSecretShareVar {
```

Comment:
is it overbearing to make a `Blinder` trait that `BalanceSecretShare` and `BalanceSecretShareVar` both implement? assuming we'll also need blinding/unblinding for all the other secret share types for other wallet components

Reply:
Not unreasonable, but that feels over-abstracted to me. It doesn't buy us any generic functionality at a higher level (these methods are always called through `Wallet::blind`) and their implementation is very simple, so I'd err on the side of readability + simplicity here